### ISSUE TITLE
Newly released lodash 4.17.0 breaks ember build

### ISSUE THREAD

**Issue by polesen** [ember-cli/ember-cli#6430]
Doing a clean `npm install` will install newly released lodash 4.17.0 under ember-cli (because `ember-cli/package.json` depends on `^4.6.1`).

I am getting this stacktrace from ember-build:
```
⠋ BuildingCannot read property '0' of undefined
TypeError: Cannot read property '0' of undefined
    at EmberApp._initVendorFiles (/share/tomcat/forsiapp/atlight-forsiapp/node_modules/ember-cli/lib/broccoli/ember-app.js:317:55)
    at new EmberApp (/share/tomcat/forsiapp/atlight-forsiapp/node_modules/ember-cli/lib/broccoli/ember-app.js:94:8)
    at module.exports (/share/tomcat/forsiapp/atlight-forsiapp/ember-cli-build.js:6:13)
    at Class.module.exports.Task.extend.setupBroccoliBuilder (/share/tomcat/forsiapp/atlight-forsiapp/node_modules/ember-cli/lib/models/builder.js:55:19)
    at Class.module.exports.Task.extend.init (/share/tomcat/forsiapp/atlight-forsiapp/node_modules/ember-cli/lib/models/builder.js:89:10)
    at new Class (/share/tomcat/forsiapp/atlight-forsiapp/node_modules/ember-cli/node_modules/core-object/core-object.js:18:12)
    at Class.module.exports.Task.extend.run (/share/tomcat/forsiapp/atlight-forsiapp/node_modules/ember-cli/lib/tasks/build.js:15:19)
    at /share/tomcat/forsiapp/atlight-forsiapp/node_modules/ember-cli/lib/commands/build.js:32:24
    at tryCatch (/share/tomcat/forsiapp/atlight-forsiapp/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:538:12)
    at invokeCallback (/share/tomcat/forsiapp/atlight-forsiapp/node_modules/ember-cli/node_modules/rsvp/dist/rsvp.js:553:13)
```

And I think it is related to to some "deep path" changes in lodash `omitBy`, because debugging a bit inside `ember-app.js` shows me, that with the new lodash the keys/properties in `vendorFiles` object has had their `.js` extension removed, which makes this line fail:

```
  if (!existsSync(this.vendorFiles['ember-testing.js'][0])) {
```

If I force lodash dependency back to `4.15.0` it works again.

Output from `ember version --verbose && npm --version`:
```
ember-cli: 2.4.3
http_parser: 2.7.0
node: 4.6.1
v8: 4.5.103.37
uv: 1.9.1
zlib: 1.2.8
ares: 1.10.1-DEV
icu: 56.1
modules: 46
openssl: 1.0.2j
os: linux x64
2.15.9
```

**Comment by rwjblue** [comment]
Hmm. Good sleuthing. Unless what we were doing was somehow unsupported, I suspect this is a bug in lodash. Have you checked and/or reported over there?

**Comment by vitch** [comment]
Yeah - it's reported [over there](https://github.com/lodash/lodash/issues/2811)

**Comment by RuslanZavacky** [comment]
And actually, latest versions of ember are not affected, because there is no lodash in package.json anymore. But as we are still on 2.5 we face this issue.

**Comment by pavlin99th** [comment]
Later versions of `ember-cli` are not affected because of this check: https://github.com/jeffjewiss/ember-cli/commit/7fb46f83d198f93790407fcae0ba89c1632d911b

**Comment by polesen** [comment]
Okay. I'm on `2.4.3` which doesn't have that fix. Must upgrade to the new 2.8 LTS then, I guess :-) Thanks for the info guys.

**Comment by nathanhammond** [comment]
Closing as it is an upstream issue which has already been fixed. Thanks all for your diligence!

**Comment by rwjblue** [comment]
FWIW, the fix isn't released as of this moment and using lodash@4.17.0 will result in errors for users of older versions of ember-cli. 

A temporary work around (to force lodash 4.16.x to be used by ember-cli) is:

```
npm install --save-dev -E lodash@4.16.6 && rm -rf node_modules && npm install
```

**Comment by allthesignals** [comment]
@rwjblue works, but seems to add glob, express, and some other dependencies that were somehow missed
